projects/power_selection/src/power_selection.c

```diff
@@ -0,0 +1,106 @@
+#include "power_selection.h"
+
+#include "adc.h"
+#include "can.h"
+#include "can_transmit.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "power_selection_events.h"
+#include "soft_timer.h"
+
+// dcdc address if on. Pull Down is means it's on
+static GpioAddress s_dcdc_address = { .port = GPIO_PORT_A, .pin = 9 };
+static GpioAddress s_aux_status_addresses[3] = {
+  { .port = GPIO_PORT_A, .pin = 0 },  // aux voltage sense
+  { .port = GPIO_PORT_A, .pin = 3 },  // aux temp sense
+  { .port = GPIO_PORT_A, .pin = 5 }   // aux current sense  //not needed in the module currently
+};
+
+static GpioSettings s_gpio_settings = {
+  GPIO_DIR_IN,        //
+  GPIO_STATE_LOW,     //
+  GPIO_RES_NONE,      //
+  GPIO_ALTFN_ANALOG,  //
+};
+// the first is voltage, the second is temp, last is current sense
+AdcChannel aux_channels[3] = { ADC_CHANNEL_0, ADC_CHANNEL_1, ADC_CHANNEL_2 };
```

Comment:
uh, i don't think they can be enums.
I use adc_get_channel, to get the proper channels since i don't know what they are. I just added random ones for now for my test.

Reply:
I mean something along the lines of
```
typedef enum { AUX_ADC_VOLT_CHANNEL = 0, AUX_ADC_TEMP_CHANNEL, AUX_ADC_CURRENT_CHANNEL, NUM_AUX_ADC_CHANNELS } AuxAdcChannel;

AdcChannel aux_channels[NUM_AUX_ADC_CHANNELS] = {
    [AUX_ADC_VOLT_CHANNEL] = ADC_CHANNEL_0,
    [AUX_ADC_TEMP_CHANNEL] = ADC_CHANNEL_1,
    etc.
}
```
then use the enum values instead of indices anywhere